### ISSUE TITLE
Rewrite the disassembler

### ISSUE THREAD

**Issue by bbrk24** [bbrk24/Trilangle#1]
Well, it's finally come to this, has it?

https://github.com/bbrk24/Trilangle/blob/e3a7a0ecae5f44fd69001caaad01fd65f17e3c60/disassembler.cpp#L7-L9

As a proof of bugginess, this program:

```
    <
   ' D
  ' - 2
 2 ' % 7
@ > { # -
```

with the `-D` flag, produces this output:

```
0:      NOP
1:      GDT
2:      PSI #2
3:      SUB
4:      PSI #7
5:      MOD
6:      PSI #2
7:      SUB
8:      NOP
9:      BNG 12
10:     DUP
11:     EXT
12:     DUP
13:     MOD
14:     TJN
15:     JMP 9
16
```

Notice the extraneous '16' after the last line.

**Comment by bbrk24** [comment]
This particular example was fixed in e0e6938, but the real problem is something much deeper.

**Comment by bbrk24** [comment]
Rewrite occurred in b3ef9a4. Huge thanks to @mousetail for guiding me on the implementation!
